src/ILCompiler.WebAssembly/src/Compiler/DependencyAnalysis/WebAssemblyVTableSlotNode.cs

```diff
@@ -44,7 +44,7 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
         {
             DependencyList result = new DependencyList();
 
-            if (!factory.VTable(_targetMethod.OwningType).HasFixedSlots)
+            if (!factory.VTable(_targetMethod.OwningType).HasFixedSlots && !_targetMethod.HasInstantiation)
```

Comment:
Let's make this an assert in the constructor (`Debug.Assert(!_targetMethod.HasInstantiation)`) and avoid adding them to the system. Generic virtual method calls are done differently - they don't go through the vtable - there's no point generating these.

Reply:
Done.  I see that's more consistent.